File: language-samples/hello.nim
// Nim sample for GitHub language detection
# BarbrickDesign - Complete Language Portfolio

type
  Greeter = object
    message: string

proc greet(g: Greeter) =
  echo g.message

proc main() =
  var greeter = Greeter(message: "Hello from Nim!")
  greeter.greet()
  
  # Seq (sequence)
  let numbers = @[1, 2, 3, 4, 5]
  var doubled = newSeq[int]()
  for num in numbers:
    doubled.add(num * 2)
  
  # Table
  import tables
  var info = initTable[string, string]()
  info["language"] = "Nim"
  info["paradigm"] = "Multi-paradigm"
  info["typing"] = "Static"
  
  # Map
  let languages = @["nim", "python", "c"]
  for lang in languages:
    echo lang.toUpperAscii()
  
  # Pattern matching
  let value = 42
  case value
  of 42:
    echo "The answer!"
  else:
    echo "Something else"
  
  echo doubled

main()

File: language-samples/hello.hs
-- Haskell sample for GitHub language detection
-- BarbrickDesign - Complete Language Portfolio

module Main where

import Data.Char (toUpper)

data Greeter = Greeter { message :: String }

greet :: Greeter -> IO ()
greet g = putStrLn (message g)

main :: IO ()
main = do
    let greeter = Greeter "Hello from Haskell!"
    greet greeter
    
    -- List comprehension
    let numbers = [1..5]
    let doubled = [x * 2 | x <- numbers]
    
    -- Map
    let languages = ["haskell", "elm", "purescript"]
    let upper = map (map toUpper) languages
    
    -- Pattern matching
    let describe x = case x of
            42 -> "The answer"
            _  -> "Something else"
    
    -- Print results
    print doubled
    mapM_ putStrLn upper
    putStrLn $ describe 42
